pkg/service/cluster/service.go

```diff
@@ -604,3 +602,18 @@ func (s *Service) notifyChangeListener(ctx context.Context, c Change) error {
 func (s *Service) Close() {
 	s.clientCache.Close()
 }
+
+func GetRPCAddresses(ctx context.Context, client *scyllaclient.Client, hosts []string) ([]string, error) {
+	var sessionHosts []string
+	var combinedError error
+	for _, h := range hosts {
+		ni, err := client.NodeInfo(ctx, h)
+		if err != nil {
+			combinedError = multierr.Append(combinedError, err)
+			continue
+		}
+		sessionHosts = append(sessionHosts, ni.CQLAddr(h))
+	}
+
+	return sessionHosts, combinedError
```

Comment:
It's often the case that the length of returned addresses is checked:
```go
sessionHosts, err := GetRPCAddresses(ctx, client, client.Config().Hosts)
	if len(sessionHosts) == 0 {
		return session, errors.Wrapf(err, "failed to build list of session ips")
	}
```
Maybe it's better to move it inside the `GetRPCAddresses`?

Reply:
actually makes sesne

Reply:
ok, I cannot move it inside GetRPCAddresses.
GetRPCAddresses returns combined errors from every client.NodeInfo invocation. 
I don't want to extend the function signature to pass logger here and log this information in the method itself.

What I did instead is I created ErrNoRPCAddressesFound which is returned (together with other ones) when there is no address found.
Then it's up to the end user to check this error and react.